packages/toolpad-app/src/toolpad/propertyControls/GridColumns.tsx

```diff
@@ -26,14 +26,26 @@ import {
   SerializableGridColumns,
 } from '@mui/toolpad-components';
 import type { EditorProps } from '../../types';
+import { useToolpadComponents } from '../AppEditor/toolpadComponents';
+import { ToolpadComponentDefinition } from '../../toolpadComponents';
+import { useDom } from '../DomLoader';
 
 // TODO: this import suggests leaky abstraction
 import { usePageEditorState } from '../AppEditor/PageEditor/PageEditorProvider';
 import { generateUniqueString } from '../../utils/strings';
 
 type GridAlignment = SerializableGridColumn['align'];
 
-const COLUMN_TYPES: string[] = ['string', 'number', 'date', 'dateTime', 'boolean', 'link', 'image'];
+const COLUMN_TYPES: string[] = [
+  'string',
+  'number',
+  'date',
+  'dateTime',
+  'boolean',
+  'link',
+  'image',
+  'customComponent',
```

Comment:
```suggestion
  'codeComponent',
```

Nomenclature so far everywhere else has been "code component"